### PR TITLE
Send all logs when going into the background

### PR DESCRIPTION
**Pull request by MatkovIvan**
Copy of #755

### CODE REVIEW

File: AppCenter/AppCenter/Internals/Channel/MSChannelDefault.m [hunk 1]
```diff
@@ -311,13 +317,54 @@ - (void)suspend {
...
+    typeof(self) strongSelf = weakSelf;
+    [strongSelf resumeSync];
+  });
+}
+
+- (void)resumeSync {
+  if (!self.sender.suspended && self.enabled) {
     MSLogDebug([MSAppCenter logTag], @"Resume channel for group Id %@.", self.configuration.groupId);
     self.suspended = NO;
+    self.stopFlushingCompletion = kMSEmptyStopFlushingCompletion;
```
Comment (jaelim-ms) [inline AppCenter/AppCenter/Internals/Channel/MSChannelDefault.m]: This shouldn't be here. Remove.

File: AppCenter/AppCenter/Internals/Channel/MSChannelDefault.m [hunk 2]
```diff
@@ -311,13 +317,54 @@ - (void)suspend {
...
     [self flushQueue];
```
Comment (jaelim-ms) [inline AppCenter/AppCenter/Internals/Channel/MSChannelDefault.m]: This is not related to this PR but I don't think this is not necessary.
BTW, if we remove this one, there is a weird behavior which background task gets invalidated with running task.

File: AppCenter/AppCenter/Internals/Channel/MSLogManagerDefault.m [hunk 3]
```diff
@@ -243,4 +383,23 @@ - (void)setLogUrl:(NSString *)logUrl {
   self.sender.baseURL = logUrl;
 }
 
+#pragma mark - Other private methods
+
+- (void)endBackgroundActivity {
+#if !TARGET_OS_OSX
+  if (!MS_IS_APP_EXTENSION) {
+    // Reset flushed channels count.
```
Comment (jaelim-ms) [inline AppCenter/AppCenter/Internals/Channel/MSLogManagerDefault.m]: Empty line before comment.

**Comment by TroubleMakerBen** [comment]
@jaelim-ms I've left a comment in VSTS.

**Comment by jaelim-ms** [comment]
After a discussion, we decide to expose an API to enable sending logs in background.

**Pull request closed without merging**